File: web/src/main/java/by/sorface/passport/web/services/users/social/TwitchOAuth2UserService.java
package by.sorface.passport.web.services.users.social;

import by.sorface.passport.web.dao.models.enums.ProviderType;
import by.sorface.passport.web.records.socialusers.TwitchOAuth2User;
import by.sorface.passport.web.services.users.RoleService;
import by.sorface.passport.web.services.users.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class TwitchOAuth2UserService extends AbstractSocialOAuth2UserService<TwitchOAuth2User>
        implements SocialOAuth2UserService<TwitchOAuth2User> {

    @Autowired
    public TwitchOAuth2UserService(final UserService userService,
                                   final RoleService roleService) {
        super(ProviderType.TWITCH, userService, roleService);
    }

}

File: backend/src/main/kotlin/by/sorface/idp/service/oauth/jdbc/TwitchOAuth2UserService.kt
package by.sorface.idp.service.oauth.jdbc

import by.sorface.idp.config.security.oauth2.records.TwitchOAuth2User
import by.sorface.idp.dao.sql.model.enums.ProviderType
import by.sorface.idp.dao.sql.repository.user.RoleRepository
import by.sorface.idp.dao.sql.repository.user.UserRepository
import by.sorface.idp.service.oauth.SocialOAuth2UserService
import org.springframework.stereotype.Service

@Service
class TwitchOAuth2UserService(
    userRepository: UserRepository,
    roleRepository: RoleRepository
) : AbstractSocialOAuth2UserService<TwitchOAuth2User>(ProviderType.TWITCH, userRepository, roleRepository), SocialOAuth2UserService<TwitchOAuth2User>